ERROR while installing ocamlfind: "ocamlfind": command not found

Running with `--debug`, I see I was actually missing `m4`, but the error wasn't very helpful:

```
bash-4.1# opam install ocamlfind
The following actions will be performed:
 - install   ocamlfind.1.5.2
=== 1 to install ===

=-=-=-=-=-=-=-=-=-=-=-=- Synchronizing package archives -=-=-=-=-=-=-=-=-=-=-=-=

=-=-=-=-=-=-=-=-=-=-=-=-=-=-= Installing packages -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
Building ocamlfind.1.5.2:
  ./configure -bindir /.opam/4.01.0/bin -sitelib /.opam/4.01.0/lib -mandir /.opam/4.01.0/man -confi.0/lib/findlib.conf
  make all
  make opt
  make install
[ERROR] The compilation of ocamlfind.1.5.2 failed.
Removing ocamlfind.1.5.2.
  ocamlfind remove bytes
  ./configure -bindir /.opam/4.01.0/bin -sitelib /.opam/4.01.0/lib -mandir /.opam/4.01.0/man -confi.0/lib/findlib.conf
  make uninstall


#=================== ERROR while installing ocamlfind.1.5.2 ===================#
Internal error:
  "ocamlfind": command not found.
```

--- a/configure.ac
+++ b/configure.ac
@@ -63,7 +63,7 @@ AC_CHECK_OCAML_PKG([re.glob])
 AC_CHECK_OCAML_PKG([cmdliner])
 AC_CHECK_OCAML_PKG([ocamlgraph])
 AC_CHECK_OCAML_PKG([cudf])
-AC_CHECK_OCAML_PKG([dose3])
+AC_CHECK_OCAML_PKG([dose3,dose])
 
 dnl echo
 dnl echo "extlib........................ ${OCAML_PKG_extlib}"

--- a/src/client/opamAction.ml
+++ b/src/client/opamAction.ml
@@ -359,11 +359,17 @@ let remove_package_aux t ~metadata ?(keep_build=false) ?(silent=false) nv =
         if not !OpamGlobals.dryrun then
           OpamFilename.exec ~env ?name exec_dir ~metadata ~keep_going:true
             remove
-      with OpamSystem.Process_error r ->
+      with
+      | OpamSystem.Process_error r ->
+          if not silent then
+            OpamGlobals.warning
+              "failure in package uninstall script, some files may remain:\n%s"
+              (OpamProcess.string_of_result r)
+      | OpamSystem.Command_not_found cmd ->
         if not silent then
           OpamGlobals.warning
-            "failure in package uninstall script, some files may remain:\n%s"
-            (OpamProcess.string_of_result r)
+            "failure in package uninstall script, some files may remain:\n%s%s"
+            cmd ": command not found"
   end;
 
   if not !OpamGlobals.dryrun then begin

--- a/src/core/opamSystem.ml
+++ b/src/core/opamSystem.ml
@@ -347,17 +347,21 @@ let commands ?verbose ?env ?name ?metadata ?(keep_going=false) commands =
   let name = log_file name in
   let run = run_process ?verbose ?env ~name ?metadata in
   let command r0 c =
-    if keep_going || OpamProcess.is_success r0 then
-      let r = run c in
-      (if OpamProcess.is_failure r0 then r0 else r)
-    else r0
+    match r0, keep_going with
+    | (`Error _ | `Exception _), false -> r0
+    | _ ->
+      let r1 = try
+          let r = run c in
+          if OpamProcess.is_success r then `Successful r else `Error r
+        with Command_not_found _ as e -> `Exception e
+      in
+      match r0 with `Start | `Successful _ -> r1 | _ -> r0
   in
-  match commands with
-  | [] -> ()
-  | c1::c ->
-    let r = List.fold_left command (run c1) c in
-    if OpamProcess.is_success r then log_cleanup r
-    else process_error r
+  match List.fold_left command `Start commands with
+  | `Start -> ()
+  | `Successful r -> log_cleanup r
+  | `Error e -> process_error e
+  | `Exception e -> raise e
 
 let read_command_output ?verbose ?env ?metadata cmd =
   let name = log_file None in

